### PR TITLE
Mod: update the comments in registy directory

### PR DESCRIPTION
**Pull request by zouyx**
**What this PR does**:

* update the comments in registy directory

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: registry/registry.go [hunk 1]
```diff
@@ -45,13 +45,21 @@ type Registry interface {
 	Subscribe(*common.URL, NotifyListener)
 }
 
-// NotifyListener ...
+// NotifyListener is used for triggered when a service change notification is received.
 type NotifyListener interface {
+	// Notify needs to support the contract: <br>
```
Comment (AlexStocks) [inline registry/registry.go:50]: so lang.

File: registry/protocol/protocol.go [hunk 2]
```diff
@@ -411,6 +416,8 @@ func newProviderConfigurationListener(overrideListeners *sync.Map) *providerConf
 	return listener
 }
 
+// Process is Listener callback method. Listener gets notified by this method once there's any change happens on the config
```
Comment (AlexStocks) [inline registry/protocol/protocol.go:419]: so long

File: registry/protocol/protocol.go [hunk 3]
```diff
@@ -435,6 +442,8 @@ func newServiceConfigurationListener(overrideListener *overrideSubscribeListener
...
```
Comment (AlexStocks) [inline registry/protocol/protocol.go:445]: so long.

File: config_center/configuration_listener.go [hunk 4]
```diff
@@ -27,6 +27,8 @@ import (
 
 // ConfigurationListener for changing listener's event
 type ConfigurationListener interface {
+	// Process is Listener callback method. Listener gets notified by this method once there's any change happens on the config
```
Comment (AlexStocks) [inline config_center/configuration_listener.go:30]: so long.

File: registry/base_configuration_listener.go [hunk 5]
```diff
@@ -29,19 +29,19 @@ import (
 	"github.com/apache/dubbo-go/remoting"
 )
 
-// BaseConfigurationListener ...
+// BaseConfigurationListener will get notified when the config it listens on changes.
```
Comment (AlexStocks) [inline registry/base_configuration_listener.go:32]: I do not think we need a comment for this struct. A "nolint" is enough.

File: registry/zookeeper/registry.go [hunk 6]
```diff
@@ -141,35 +142,43 @@ func (r *zkRegistry) InitListeners() {
...
 	r.client.Close()
 	r.client = nil
 }
 
+// ZkClient gets zookeeper client
 func (r *zkRegistry) ZkClient() *zookeeper.ZookeeperClient {
 	return r.client
 }
 
+// SetZkClient sets zookeeper client
```
Comment (AlexStocks) [inline registry/zookeeper/registry.go:171]: a `nolint` is enough.

File: registry/zookeeper/registry.go [hunk 7]
```diff
@@ -141,35 +142,43 @@ func (r *zkRegistry) InitListeners() {
...
 	return r.getListener(conf)
 }
 
+// CloseAndNilClient closes listeners and clear client
 func (r *zkRegistry) CloseAndNilClient() {
 	r.client.Close()
 	r.client = nil
 }
 
+// ZkClient gets zookeeper client
```
Comment (AlexStocks) [inline registry/zookeeper/registry.go:166]: a `nolint` is enough.

File: registry/zookeeper/registry.go [hunk 8]
```diff
@@ -141,35 +142,43 @@ func (r *zkRegistry) InitListeners() {
...
 func (r *zkRegistry) ZkClient() *zookeeper.ZookeeperClient {
 	return r.client
 }
 
+// SetZkClient sets zookeeper client
 func (r *zkRegistry) SetZkClient(client *zookeeper.ZookeeperClient) {
 	r.client = client
 }
 
+// ZkClientLock gets zookeeper registry control lock
```
Comment (AlexStocks) [inline registry/zookeeper/registry.go:176]: a `nolint` is enough.

File: registry/zookeeper/registry.go [hunk 9]
```diff
@@ -141,35 +142,43 @@ func (r *zkRegistry) InitListeners() {
...
 func (r *zkRegistry) ZkClientLock() *sync.Mutex {
 	return &r.cltLock
 }
 
+// CloseListener closes listeners
```
Comment (AlexStocks) [inline registry/zookeeper/registry.go:181]: a `nolint` is enough.

File: registry/registry.go [hunk 10]
```diff
@@ -45,13 +45,16 @@ type Registry interface {
 	Subscribe(*common.URL, NotifyListener)
 }
 
-// NotifyListener ...
+// NotifyListener is used for triggered when a service change notification is received.
```
Comment (AlexStocks) [inline registry/registry.go:48]: a `nolint` is enough.

File: registry/nacos/listener.go [hunk 11]
```diff
@@ -212,6 +214,7 @@ func (nl *nacosListener) Next() (*registry.ServiceEvent, error) {
 	}
 }
 
+// Close nacos registry center
```
Comment (AlexStocks) [inline registry/nacos/listener.go:217]: a `nolint` is enough.

File: config_center/configuration_listener.go [hunk 12]
```diff
@@ -27,6 +27,7 @@ import (
 
 // ConfigurationListener for changing listener's event
 type ConfigurationListener interface {
+	// Process notified once there's any change happens on the config
```
Comment (AlexStocks) [inline config_center/configuration_listener.go:30]: Process the notification event once there's any change happens on the config

File: registry/base_configuration_listener.go [hunk 13]
```diff
@@ -29,19 +29,19 @@ import (
...
 }
 
-// Configurators ...
+// Configurators gets Configurator from config center
 func (bcl *BaseConfigurationListener) Configurators() []config_center.Configurator {
 	return bcl.configurators
 }
 
-// InitWith ...
+// InitWith will init BaseConfigurationListener with @key 、@listener and @f
```
Comment (AlexStocks) [inline registry/base_configuration_listener.go:44]: InitWith will init BaseConfigurationListener by @key+@listener+@f

File: registry/base_configuration_listener.go [hunk 14]
```diff
@@ -82,14 +82,14 @@ func (bcl *BaseConfigurationListener) genConfiguratorFromRawRule(rawConfig strin
 	return nil
 }
 
-// OverrideUrl ...
+// OverrideUrl gets existing configuration rule and override provider url before exporting.
```
Comment (AlexStocks) [inline registry/base_configuration_listener.go:85]: OverrideUrl gets existing configuration rule and overrides provider url before exporting.

File: registry/base_configuration_listener.go [hunk 15]
```diff
@@ -60,7 +60,7 @@ func (bcl *BaseConfigurationListener) InitWith(key string, listener config_cente
 	}
 }
 
-// Process ...
+// Process can reference ConfigurationListener.Process
```
Comment (AlexStocks) [inline registry/base_configuration_listener.go:63]: "Process can reference ConfigurationListener.Process" I can not get what the meaning of it.

File: registry/base_configuration_listener.go [hunk 16]
```diff
@@ -82,14 +82,14 @@ func (bcl *BaseConfigurationListener) genConfiguratorFromRawRule(rawConfig strin
...
-// OverrideUrl ...
+// OverrideUrl gets existing configuration rule and override provider url before exporting.
 func (bcl *BaseConfigurationListener) OverrideUrl(url *common.URL) {
 	for _, v := range bcl.configurators {
 		v.Configure(url)
 	}
 }
 
-// ToConfigurators ...
+// ToConfigurators converts override urls to map for use when re-refer. Send all rules every time, the urls will be reassembled and  calculated
```
Comment (AlexStocks) [inline registry/base_configuration_listener.go:92]: ToConfigurators converts override urls to map for use when re-refer. Send all rules every time, the urls will be reassembled and  calculated

把汉语句子发给我，我来改。

File: registry/consul/listener.go [hunk 17]
```diff
@@ -187,6 +187,7 @@ func (l *consulListener) handler(idx uint64, raw interface{}) {
 	}
 }
 
+// Next returns next service event once received
```
Comment (AlexStocks) [inline registry/consul/listener.go:190]: Next returns the service event from consul.

File: registry/consul/registry.go [hunk 18]
```diff
@@ -95,6 +95,7 @@ func (r *consulRegistry) register(url common.URL) error {
 	return r.client.Agent().ServiceRegister(service)
 }
 
+// Unregister service from consul registry center
```
Comment (AlexStocks) [inline registry/consul/registry.go:98]: Unregister deletes service from consul registry center

File: registry/consul/registry.go [hunk 19]
```diff
@@ -156,10 +158,12 @@ func (r *consulRegistry) getListener(url common.URL) (registry.Listener, error)
 	return listener, err
 }
 
+// GetUrl get registry URL of consul registry center
 func (r *consulRegistry) GetUrl() common.URL {
 	return *r.URL
 }
 
+// IsAvailable determines consul registry center whether is available
```
Comment (AlexStocks) [inline registry/consul/registry.go:166]: IsAvailable checks whether the consul registry center is available or not.

File: registry/etcdv3/listener.go [hunk 20]
```diff
@@ -38,15 +38,17 @@ type dataListener struct {
 	listener      config_center.ConfigurationListener
 }
 
-// NewRegistryDataListener
+// NewRegistryDataListener creates a data listener for etcd
 func NewRegistryDataListener(listener config_center.ConfigurationListener) *dataListener {
 	return &dataListener{listener: listener}
 }
 
+// AddInterestedURL add more URL of registry center to listen
```
Comment (AlexStocks) [inline registry/etcdv3/listener.go:46]: AddInterestedURL adds a registration @url to @l.

File: registry/etcdv3/listener.go [hunk 21]
```diff
@@ -38,15 +38,17 @@ type dataListener struct {
...
 func (l *dataListener) AddInterestedURL(url *common.URL) {
 	l.interestedURL = append(l.interestedURL, url)
 }
 
+// Process data change event from registry center of etcd
 func (l *dataListener) DataChange(eventType remoting.Event) bool {
```
Comment (AlexStocks) [inline registry/etcdv3/listener.go:52]: DataChange processes the data change event from registry center of etcd

File: registry/event.go [hunk 22]
```diff
@@ -36,7 +36,7 @@ func init() {
 // service event
 // ////////////////////////////////////////
 
-// ServiceEvent ...
+// ServiceEvent is create、update or delete event of service
```
Comment (AlexStocks) [inline registry/event.go:39]: ServiceEvent includes create, update, delete event

File: registry/event_listener.go [hunk 23]
```diff
@@ -36,6 +36,7 @@ type ConditionalEventListener interface {
 	Accept(e Event) bool
 }
 
+// ServiceInstancesChangedListener is use for the Service Discovery Changed
```
Comment (AlexStocks) [inline registry/event_listener.go:39]: ServiceInstancesChangedListener is used when the Service Discovery Changed

File: registry/kubernetes/listener.go [hunk 24]
```diff
@@ -38,12 +38,12 @@ type dataListener struct {
...
 }
 
-// NewRegistryDataListener
+// NewRegistryDataListener creates a data listener for kubernetes
 func NewRegistryDataListener(listener config_center.ConfigurationListener) *dataListener {
 	return &dataListener{listener: listener}
 }
 
-// AddInterestedURL
+// AddInterestedURL add more URL of registry center to listen
```
Comment (AlexStocks) [inline registry/kubernetes/listener.go:46]: AddInterestedURL adds the @url of registry center to the listener

File: registry/kubernetes/listener.go [hunk 25]
```diff
@@ -91,10 +91,12 @@ func NewConfigurationListener(reg *kubernetesRegistry) *configurationListener {
 	return &configurationListener{registry: reg, events: make(chan *config_center.ConfigChangeEvent, 32)}
 }
 
+// Process data change event from config center of kubernetes
```
Comment (AlexStocks) [inline registry/kubernetes/listener.go:94]: Process processes the data change event from config center of kubernetes

File: registry/mock_registry.go [hunk 26]
```diff
@@ -30,13 +30,13 @@ import (
 	"github.com/apache/dubbo-go/common/logger"
 )
 
-// MockRegistry ...
+// MockRegistry is use for mock registry
```
Comment (AlexStocks) [inline registry/mock_registry.go:33]: MockRegistry is used as mock registry

File: registry/mock_registry.go [hunk 27]
```diff
@@ -30,13 +30,13 @@ import (
...
 type MockRegistry struct {
 	listener  *listener
 	destroyed *atomic.Bool
 }
 
-// NewMockRegistry ...
+// NewMockRegistry  is use for create a mock registry
```
Comment (AlexStocks) [inline registry/mock_registry.go:39]: NewMockRegistry creates a mock registry

File: registry/mock_registry.go [hunk 28]
```diff
@@ -46,23 +46,23 @@ func NewMockRegistry(url *common.URL) (Registry, error) {
 	return registry, nil
 }
 
-// Register ...
+// Register is use for register a mock registry
```
Comment (AlexStocks) [inline registry/mock_registry.go:49]: Register is used as a mock registry

File: registry/mock_registry.go [hunk 29]
```diff
@@ -46,23 +46,23 @@ func NewMockRegistry(url *common.URL) (Registry, error) {
...
 func (*MockRegistry) Register(url common.URL) error {
 	return nil
 }
 
-// Destroy ...
+// Destroy is use for destory a mock registry
```
Comment (AlexStocks) [inline registry/mock_registry.go:54]: nolint

File: registry/mock_registry.go [hunk 30]
```diff
@@ -46,23 +46,23 @@ func NewMockRegistry(url *common.URL) (Registry, error) {
...
 }
 
-// IsAvailable ...
+// IsAvailable is use for determine a mock registry available
 func (r *MockRegistry) IsAvailable() bool {
 	return !r.destroyed.Load()
 }
 
-// GetUrl ...
+// GetUrl is use for register a mock registry URL
```
Comment (AlexStocks) [inline registry/mock_registry.go:65]: nolint

File: registry/mock_registry.go [hunk 31]
```diff
@@ -71,7 +71,7 @@ func (r *MockRegistry) subscribe(*common.URL) (Listener, error) {
 	return r.listener, nil
 }
 
-// Subscribe ...
+// Subscribe is use for subscribe a mock registry
```
Comment (AlexStocks) [inline registry/mock_registry.go:74]: nolint

File: registry/mock_registry.go [hunk 32]
```diff
@@ -123,7 +123,7 @@ func (*listener) Close() {
 
 }
 
-// MockEvent ...
+// MockEvent is use for register a mock event
```
Comment (AlexStocks) [inline registry/mock_registry.go:126]: nolint

File: registry/nacos/listener.go [hunk 33]
```diff
@@ -109,6 +109,7 @@ func generateUrl(instance model.Instance) *common.URL {
 	)
 }
 
+// Callback will callback when subscribed
```
Comment (AlexStocks) [inline registry/nacos/listener.go:112]: Callback will be invoked when got subscribed events.

File: registry/nacos/listener.go [hunk 34]
```diff
@@ -198,6 +199,7 @@ func (nl *nacosListener) process(configType *config_center.ConfigChangeEvent) {
 	nl.events <- configType
 }
 
+// Next returns next service event once received
```
Comment (AlexStocks) [inline registry/nacos/listener.go:202]: Next returns the service event from nacos.

File: registry/nacos/registry.go [hunk 35]
```diff
@@ -123,6 +123,7 @@ func createRegisterParam(url common.URL, serviceName string) vo.RegisterInstance
 	return instance
 }
 
+// Register service to nacos registry center
```
Comment (AlexStocks) [inline registry/nacos/registry.go:126]: Register will register the service @url to its nacos registry center

File: registry/nacos/registry.go [hunk 36]
```diff
@@ -174,14 +175,17 @@ func (nr *nacosRegistry) Subscribe(url *common.URL, notifyListener registry.Noti
 	}
 }
 
+// GetUrl get registry URL of nacos registry center
```
Comment (AlexStocks) [inline registry/nacos/registry.go:178]: GetUrl gets its registration URL

File: registry/nacos/registry.go [hunk 37]
```diff
@@ -174,14 +175,17 @@ func (nr *nacosRegistry) Subscribe(url *common.URL, notifyListener registry.Noti
...
 func (nr *nacosRegistry) GetUrl() common.URL {
 	return *nr.URL
 }
 
+// IsAvailable determines nacos registry center whether is available
```
Comment (AlexStocks) [inline registry/nacos/registry.go:183]: IsAvailable determines nacos registry center whether it is available

File: registry/nacos/registry.go [hunk 38]
```diff
@@ -174,14 +175,17 @@ func (nr *nacosRegistry) Subscribe(url *common.URL, notifyListener registry.Noti
...
 func (nr *nacosRegistry) IsAvailable() bool {
 	return true
 }
 
+// Destroy nacos registry center
```
Comment (AlexStocks) [inline registry/nacos/registry.go:188]: nolint

File: registry/protocol/protocol.go [hunk 39]
```diff
@@ -229,6 +231,7 @@ func newOverrideSubscribeListener(overriderUrl *common.URL, invoker protocol.Inv
 	return &overrideSubscribeListener{url: overriderUrl, originInvoker: invoker, protocol: proto}
 }
 
+// Notify will triggered when a service change notification is received.
```
Comment (AlexStocks) [inline registry/protocol/protocol.go:234]: Notify will be triggered when a service change notification is received.

File: registry/zookeeper/registry.go [hunk 40]
```diff
@@ -116,6 +116,7 @@ func newMockZkRegistry(url *common.URL, opts ...zookeeper.Option) (*zk.TestClust
 	return c, r, nil
 }
 
+// InitListeners init listeners of zookeeper registry center
```
Comment (AlexStocks) [inline registry/zookeeper/registry.go:119]: InitListeners initializes listeners of zookeeper registry center

File: registry/zookeeper/registry.go [hunk 41]
```diff
@@ -141,35 +142,43 @@ func (r *zkRegistry) InitListeners() {
 	r.dataListener = newDataListener
 }
 
+// CreatePath create the path in the registry center of zookeeper
```
Comment (AlexStocks) [inline registry/zookeeper/registry.go:145]: creates

File: registry/zookeeper/registry.go [hunk 42]
```diff
@@ -141,35 +142,43 @@ func (r *zkRegistry) InitListeners() {
...
 func (r *zkRegistry) CreatePath(path string) error {
 	return r.ZkClient().Create(path)
 }
 
+// DoRegister actually do the register job in the registry center of zookeeper
 func (r *zkRegistry) DoRegister(root string, node string) error {
 	return r.registerTempZookeeperNode(root, node)
 }
 
+// DoSubscribe actually subscribe the provider URL
```
Comment (AlexStocks) [inline registry/zookeeper/registry.go:155]: subscribes

File: registry/consul/registry.go (hunk 18 above)
Comment (zouyx) [reply registry/consul/registry.go:98]: I think `Unregister service from consul registry center` is a full sentence.

**Pull request merged**